sdk/batch/pom.xml

```diff
@@ -0,0 +1,37 @@
+<!-- Copyright (c) Microsoft Corporation. All rights reserved.
+     Licensed under the MIT License. -->
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+  <groupId>com.azure</groupId>
+  <artifactId>azure-batch-parent</artifactId>
+  <packaging>pom</packaging>
+  <version>1.0.0</version><!-- Need not change for every release-->
+
+  <name>Microsoft Azure Batch SDK Parent</name>
+  <description>Java library for Azure Batch SDK Parent</description>
+  <url>https://github.com/Azure/azure-sdk-for-java</url>
+
+  <parent>
+    <groupId>com.azure</groupId>
+    <artifactId>azure-data-sdk-parent</artifactId>
+    <version>1.0.0</version>
+    <relativePath>../../pom.data.xml</relativePath>
+  </parent>
+
+  <distributionManagement>
+    <site>
+      <id>azure-java-build-docs</id>
+      <url>${site.url}/site/${project.artifactId}</url>
+    </site>
+  </distributionManagement>
+
+  <scm>
+    <url>scm:git:https://github.com/Azure/azure-sdk-for-java</url>
+  </scm>
+
+  <modules>
+    <module>azure-batch</module>
+  </modules>
+</project>
```

Comment:
I don't understand why this file needs to be added? It specifies a groupId of com.azure, which is our track two group Id, and configures its parent to be the track one parent pom. This is then referenced by the track one azure-batch module, which should be renamed to microsoft-azure-batch based on @weshaggard's guidelines. You would then have the pom.data.xml file module reference point to ./sdk/batch/microsoft-azure-batch.

Reply:
I might have the POMs setup incorrectly in this instance, let me explain what we are trying to achieve from a CI perspective an it might make more sense (and then we can figure out exactly what the group Ids need to be.

For each service we want to have a folder (e.g. ```sdk/batch``` and ```sdk/servicebus```). For the optimized PR/CI pipelines our pipelines will just target the ```pom.xml``` file in each service directory to trigger the build).

So that means that if anything changes in ```sdk/batch/*``` where we might have both track 1 and track 2 libraries we'll build the whole set under that path.

Reply:
You will have difficulty introducing a single pom.xml file per service, as each track has a different parent that needs to be used.

Reply:
Arguably we have the same issue with ```sdk/servicebus``` where we'll have track 1 and track 2 libraries as well.

Reply:
Yes, the same issue exists there as well.

Reply:
Actually looking through the POM files now, here are the mappings to group IDs:

| POM | Parent POM | Group ID |
| - | - | - |
| ```parent/pom.xml``` | N/A | ```com.azure``` |
| ```pom.xml``` | ```parent/pom.xml``` | ```com.azure``` |
| ```pom.management.xml``` | ```parent/pom.xml``` | ```com.microsoft.azure``` |
| ```pom.data.xml``` | ```parent/pom.xml``` | ```com.azure``` |
| ```pom.client.xml``` | ```parent/pom.xml``` | ```com.azure``` |

So it seems that with the exception of the management libraries the parent POMs tend to have the group ID of ```com.azure``` even when child modules they reference (in track 1 scenarios) might be ```com.microsoft.azure```.

With regards to your comment about ```azure-batch``` becoming ```microsoft-azure-batch``` - yes that needs to be done. My primary goal for putting this out for review was to pick up these issues but also make sure the POM file structure was correct (or at least start a conversation about it).

Mission accomplished :)

Reply:
In my opinion the pom.data.xml groupID is incorrect, it should be com.microsoft.azure. Only track two would be in com.azure. @jianghaolu, do you agree?

Reply:
Do we publish the parent POM files anywhere? I'm wondering whether it matters what their groupId is. In terms of the longer term solution for how we can have a mix of track 1, track 2 and management libraries in the same ```sdk/[servicename]``` folder, I'm wondering whether profiles would help?

Reply:
Yes, parent pom files are published just like any other Maven release.... Just without any Java jar files.

Reply:
OK - I had a chat to Wes this morning to help figure out the best approach here. I think where I've gone wrong here is that I've pointed the ```pom.xml``` files for each of the libraries to the ```pom.xml``` in the service directory, whereas what I should do is just but from the service directory ```pom.xml``` to the libraries as an aggregator. The library POMs would then point to the appropriate parent in the root. I'll make this changes to the PR.

Reply:
Yes we not reparent to the pom file in the service folder, it should just be for aggregation purposes. In fact we should probably naming it something different as well just to be clear that it is intended for the building the service directory. Perhaps we call it "eng.pom.xml" or "service.pom.xml"?

Reply:
Done in the latest commit. I also adjusted it for servicebus as well since I needed to change the ```archetype-sdk-client.yml``` to reference ```pom.service.xml``` in each service directory.

Reply:
OK - I think this is where it needs to be.